compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -114,7 +116,65 @@ impl<'a> Lexer<'a> {
         }
     }
 
+    fn next_annotation_token(&mut self) -> SpannedTokenResult {
+        match self.next_char() {
+            Some('\n') => { 
+                self.fv_annotation = false;
+                self.single_char_token(Token::EOL)
+            },
+            Some(x) if Self::is_code_whitespace(x) || x == '\\' => {
+                let spanned = self.eat_whitespace_and_backslash(x);
+                if self.skip_whitespaces {
+                    self.next_token()
+                } else {
+                    Ok(spanned)
+                }
+            }
+            Some('<') => self.glue(Token::Less),
+            Some('>') => self.glue(Token::Greater),
+            Some('=') => self.glue(Token::Assign),
+            Some('/') => self.glue(Token::Slash),
+            Some('.') => self.glue(Token::Dot),
+            Some(':') => self.glue(Token::Colon),
+            Some('!') => self.glue(Token::Bang),
+            Some('-') => self.glue(Token::Minus),
+            Some('&') => self.ampersand(),
+            Some('|') => self.single_char_token(Token::Pipe),
+            Some('%') => self.single_char_token(Token::Percent),
+            Some('^') => self.single_char_token(Token::Caret),
+            Some(';') => self.single_char_token(Token::Semicolon),
+            Some('*') => self.single_char_token(Token::Star),
+            Some('(') => self.single_char_token(Token::LeftParen),
+            Some(')') => self.single_char_token(Token::RightParen),
+            Some(',') => self.single_char_token(Token::Comma),
+            Some('+') => self.single_char_token(Token::Plus),
+            Some('{') => self.single_char_token(Token::LeftBrace),
+            Some('}') => self.single_char_token(Token::RightBrace),
+            Some('[') => self.single_char_token(Token::LeftBracket),
+            Some(']') => self.single_char_token(Token::RightBracket),
+            Some('$') => self.single_char_token(Token::DollarSign),
+            Some('"') => self.eat_string_literal(),
+            Some('f') => self.eat_format_string_or_alpha_numeric(),
+            Some('r') => self.eat_raw_string_or_alpha_numeric(),
+            Some('q') => self.eat_quote_or_alpha_numeric(),
+            Some('#') => self.eat_attribute(),
+            Some(ch) if ch.is_ascii_alphanumeric() || ch == '_' => self.eat_alpha_numeric(ch),
+            Some(ch) => {
+                // We don't report invalid tokens in the source as errors until parsing to
+                // avoid reporting the error twice. See the note on Token::Invalid's documentation for details.
+                Ok(Token::Invalid(ch).into_single_span(self.position))
+            }
+            None => {
+                self.done = true;
+                Ok(Token::EOF.into_single_span(self.position))
+            }
+        }
+    }
+
     fn next_token(&mut self) -> SpannedTokenResult {
+        if self.fv_annotation{
```

Comment:
If we drop the EOL token, we could drop this function call and all of the code that it brings along; so the lexer will be simpler, even though the parser will be more complicated.

I think it would be the more standard design.